objects/src/accounts/storage/mod.rs

```diff
@@ -227,6 +273,18 @@ impl AccountStorage {
             self.set_item(slot_idx, slot_value)?;
         }
 
+        for &(slot_idx, ref map_delta) in delta.updated_maps.iter() {
+            // layout commitment slot cannot be updated
+            if slot_idx == Self::SLOT_LAYOUT_COMMITMENT_INDEX {
+                return Err(AccountError::StorageSlotIsReserved(slot_idx));
+            }
+
+            // pick the right storage map and apply delta
+            let storage_map =
+                self.maps.get_mut(&slot_idx).ok_or(AccountError::StorageMapNotFound(slot_idx))?;
+            storage_map.apply_delta(map_delta.clone())?;
```

Comment:
If we change `StorageMap::apply_delta()` to take `delta` as a reference, we won't need this `clone()` here.

Reply:
indeed